portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -2,40 +2,39 @@
 * Purpose: contains various JQuery functions that listen for events on main portfolio page
 */
 
-import { update_background, NUM_IMAGES } from './background_functions.js';
+import { updateBackground, NUM_IMAGES } from './background_functions.js';
+import { preloadImages, detectMobileDevice } from './helper_functions.js';
 
-let profile_image_paths = ["images/noogler.gif", "images/zoebakercircle.png"];
+let profileImagePaths = ["images/noogler.gif", "images/zoebakercircle.png"];
 
+let currentImageIndex = 1; // start on second image so as not to repeat landing image
 
-// Loads image paths into an arbitrary array, in order to preload images for improved performance.
-function preload_images(image_paths){
-  let image_arr = new Array(image_paths.length);
-  for (let i = 0; i < image_paths.length; i++) {
-    let temp_image = new Image(); 
-    temp_image.src = image_paths[i]; 
-    image_arr[i] = temp_image;
-  }
-  return image_arr;
-}
-
-let current_image_index = 1; // start on second image so as not to repeat landing image
 $(document).ready(function(){
+  let isMobile = detectMobileDevice();
+
+  let iconImages = preloadImages(profileImagePaths);
```

Comment:
Do we need to preloadImages on mobile? They can't change the background right? Doesn't this make them load more data then necessary on a mobile device?

Reply:
For this case, preloadImages is used only for changing the icon photo (not the background), and that functionality is implemented in mobile.